system/src/commonMain/kotlin/com/splendo/kaluga/system/network/state/NetworkStateRepo.kt

```diff
@@ -33,19 +33,22 @@ class NetworkStateRepo(
         fun create(): NetworkStateRepo
     }
 
-    private var _lastKnownNetwork = AtomicReference<Network>(Network.Unknown.WithoutLastNetwork(Network.Unknown.Reason.NOT_CLEAR))
+    private val _lastKnownNetwork = AtomicReference<Network>(Network.Unknown.WithoutLastNetwork(Network.Unknown.Reason.NOT_CLEAR))
     internal var lastKnownNetwork: Network
         get() = _lastKnownNetwork.get()
         set(value) { _lastKnownNetwork.set(value) }
 
-    internal var networkManager: BaseNetworkManager? = null
+    private val _networkManager = AtomicReference<BaseNetworkManager?>(null)
+    internal var networkManager: BaseNetworkManager?
```

Comment:
@thoutbeckers Tests were not passing due to `networkManager` that was not a AtomicReference and couldn't be changed from test thread. Is it okay to keep it as AtomicReference?

Reply:
Why is it set from the test thread?

Reply:
Because when the test run we are not in main I think

Reply:
I guess this makes sense sense.